### PR TITLE
Develop/jerbot reformation

### PR DESCRIPTION
**Pull request by lujeremy**
- An update to most of the current commands: exception handling, fixed broken apis etc.
- Updates to pom for additional dependencies while building
- Addition of a !help command to guide users through the bot

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -29,6 +29,20 @@
             <version>3.11.0</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.json</groupId>
+            <artifactId>json</artifactId>
+            <version>20180813</version>
+        </dependency>
+
```
Comment (vincentzhang96) [inline pom.xml]: nit: double newline -> single

File: pom.xml [hunk 2]
```diff
@@ -29,6 +29,20 @@
...
+
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.15</version>
+        </dependency>
+
+
```
Comment (vincentzhang96) [inline pom.xml]: ditto

File: src/main/java/io/jlu/jerbot/JerBot.java [hunk 3]
```diff
@@ -51,25 +53,24 @@ public void onMessageReceived(MessageReceivedEvent event) {
...
         }
 
         String[] contentArr = contentRaw.split(" ", 2);
         String command = contentArr[0].substring(1).toLowerCase();
         String parameter = "";
+
         if (contentArr.length > 1) {
             parameter = contentArr[1].toLowerCase();
         }
-
```
Comment (vincentzhang96) [inline src/main/java/io/jlu/jerbot/JerBot.java]: nit: newline after closing curly (style choice tho)

File: src/main/java/io/jlu/jerbot/commands/HelpCommand.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                    definition = "Say hello!";
+                    break;
+                case "record":
+                    definition = "Record your workout. Format is 'workout reps weight,' e.g. Bench press 5 100";
+                    break;
+                case "show":
+                    definition = "Shows all of the workouts you've recorded thus far";
+                    break;
+            }
+//            System.out.println("I just added: " + name + definition);
```
Comment (vincentzhang96) [inline src/main/java/io/jlu/jerbot/commands/HelpCommand.java]: Remove commented out debug console output

File: src/main/java/io/jlu/jerbot/commands/HelpCommand.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class HelpCommand implements Command {
+
+    Map<String, String> commandDefinitions;
+
+    public HelpCommand() {
+        this.commandDefinitions = defineCommands();
```
Comment (lujeremy) [inline src/main/java/io/jlu/jerbot/commands/HelpCommand.java]: Here I'm declaring a separate map in defineCommands(), and then setting our class field to equal that map (In this case is it just a shift in variable reference and therefore won't be costly at all?). Wasn't sure if it is better to define the class field as a new map, and then **directly** .put() information into it.

**Pull request merged**